chromium_src/chrome/renderer/worker_content_settings_client.cc

```diff
@@ -85,4 +101,19 @@ bool WorkerContentSettingsClient::HasContentSettingsRules() const {
   return content_setting_rules_.get();
 }
 
+#define BRAVE_WORKER_CONTENT_SETTINGS_CLIENT                                   \
+  if (const auto& shields_settings =                                           \
+          static_cast<content_settings::BraveContentSettingsAgentImpl*>(agent) \
+              ->shields_settings()) {                                          \
+    shields_settings_ = shields_settings->Clone();                             \
```

Comment:
why isn't shields_settings part of RendererContentSettingRules?

Reply:
I'm also a bit confused about why we'd have things in ContentSettingsAgent that aren't content settings like reduce langage and origins to allow scripts and then take something that is a content setting (farbling level) and treat it differently.

Reply:
I'm in favor of something that gets the "effective" setting, but I'm not sure that ContentSettingsAgent is the right place for this and maybe we should either use `RendererUpdater` or create something new. There are a lot of problems with the farbling content setting (like using CONTENT_SETTINGS_DEFAULT as a distinct value) and the current issue for most settings that we always have to check shields in addition to the content setting. I've been thinking that similar to cookies we should have a real content setting (BRAVE_COOKIES) and an effective content setting (COOKIES) that gives you the value after applying shields.

Reply:
This does make sense as part of WebContentSettingsClient, but I'm not a fan of adding this to ContentSettingsAgent/RendererContentSettingRules by adding a new mojo type for shield settings

Reply:
> why isn't shields_settings part of RendererContentSettingRules?

mainly because of the patches. This is not the usual `ContentSettingsForOneType` list that's just prefilled with values from `HostContentSettingsMap`. To support this as a part of `RendererContentSettingRules` we will need to add some patches into `PageSpecificContentSettings`, which we can avoid entirely.

I also need to use this structure for the proper workers farbling. Right now we only have this:
https://github.com/brave/brave-core/blob/09558152d805449ce0928e6837cf93ecff3b93d0/chromium_src/third_party/blink/public/mojom/worker/worker_content_settings_proxy.mojom#L9-L14

> I'm also a bit confused about why we'd have things in ContentSettingsAgent that aren't content settings like reduce langage and origins to allow scripts and then take something that is a content setting (farbling level) and treat it differently.

we may look into creating a separate `BraveShieldsSettings` class on the blink side, but for now I think it's acceptable to keep it as part of `ContentSettingsAgent`. It seems like too much plumbing will be required to introduce something like `ContentSettingsAgent` and wire it will all users in the renderer.

> I'm in favor of something that gets the "effective" setting

yep, one of the end goals is to calculate the effective shields configuration before passing it to the renderer, but this is out of the scope of this PR currently.

Reply:
> It seems like too much plumbing will be required to introduce something like ContentSettingsAgent and wire it will all users in the renderer.

`Supplement<ExecutionContext>` should make it available anywhere that ContentSettingsAgent is available just like we have done in other places like `BraveSessionCache`? I'm fine following up with something like this and also https://github.com/brave/brave-core/pull/25505#discussion_r1763824924 but I think it's something we should do because this seems a bit out of place to me.